vendor/brave-ios/Ledger/Data/BATLedgerDatabase.mm

```diff
@@ -990,4 +994,178 @@ + (void)insertOrUpdateLinksForServerPublisherInfo:(ServerPublisherInfo *)info
   }
 }
 
+#pragma mark - Promotions
+
++ (nullable Promotion *)getPromotionWithID:(NSString *)promoID context:(NSManagedObjectContext *)context
+{
+  return [self firstOfClass:Promotion.class
+                 predicates:@[[NSPredicate predicateWithFormat:@"promotionID == %@", promoID]]
```

Comment:
cc @kylehickinson

Reply:
@masparrow  This is not usually standard practice for Obj-C/CoreData as we have no way of un-stringifying model variables because CoreData models exist in a non-code accessible file. This would end up with me creating single use string const's at the top of the file

Reply:
If the are all single use, then no worries - this was probably codeblindness from seeing similar  everywhere else (outside of iOS I mean)